src/Chapter3.hs

```diff
@@ -970,6 +1058,21 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data WeekDay = Sunday|Monday|Tuesday|Wednesday|Thursday|Friday|Saturday deriving (Show,Read,Eq,Ord,Enum)
+
+isWeekend::WeekDay->Bool
+isWeekend d 
+  | d == Sunday || d == Saturday = True
+  | otherwise = False
+
+nextDay :: WeekDay->WeekDay
+nextDay Saturday = Sunday
```

Comment:
That is right!
You could also use the instance of `Bounded` in here:

```suggestion
    | day == maxBound = minBound
```

This is also a bit more maintainable. If you decide, for example, to have the "European" system of week days in the data type constructors, you won't need to change this function at all! 🙂